### DISCUSSION TITLE
How to get the left side material properties value in DG kernel

### DISCUSSION THREAD

**Discussion by SomnusYu** [idaholab/moose#26663] [Q&A General]
Hi all,

I'm solving a 1-D problem in ADDG kernel and I need to get the material properties value at left element and right element. I know that we can get the material properties value at the right element by using `getNeighborADMaterialProperty` (for current element value using `getADMaterialProperty`). I would like to know how to get the material properties value at the left element.

Thanks!

**Comment by GiudGiud** [comment]
Hello

DGKernels act on quadrature points on internal side. So there are only two elements to consider: the current element and the neighbor. You cannot be considering at a single quadrature point both the current, the left and the right neighbors. 

See the docs for more info
https://mooseframework.inl.gov/syntax/DGKernels/index.html

What are you trying to do here? Get a spatial derivative with 3 element values?

Guillaume

**Reply by SomnusYu** [reply]
Yes, I would like to calculate a spatial derivative: `∂alpha/∂x = (alpha_i+1/2 - alpha_i-1/2)/Δx = (alpha_i+1 - alpha_i-1)/(2Δx)` So it needs to get alpha_i+1(right element) and alpha_i-1(left element).

**Reply by GiudGiud** [reply]
Could you use a non-centered spatial derivative?
On the right do: (alpha_right - alpha_current-elem) / dx

**Reply by SomnusYu** [reply]
> On the right do: (alpha_right - alpha_current-elem) / dx

In this way, I just need to calculate the spatial derivative in DGKernel by: 
```
    case Moose::Element:
        re = alpha_current-elem * _test[_i][_qp];
        break;

    case Moose::Neighbor:
        re = -alpha_right * _test_neighbor[_i][_qp];
        break;

```

Is that right?

**Reply by GiudGiud** [reply]
This is for the residual?
What is the strong form of this equation term?

**Reply by SomnusYu** [reply]
The equation is a little complex.
![image](https://github.com/idaholab/moose/assets/118275988/ae8f4fe7-2146-410f-8d1c-d0a9d7e17392)
![image](https://github.com/idaholab/moose/assets/118275988/2718c62e-cc0e-46bd-9a25-4c4c45bdb25c)
![image](https://github.com/idaholab/moose/assets/118275988/5cf97e23-923b-4a5f-b31a-4a5a56680b3c)

I have already finished the conserved term:`∂F/∂x`. I still need to construct the non-conserved term, that's `∂alpha/∂x`.

**Reply by GiudGiud** [reply]
This looks right to me.
Unless you want to evaluate alpha on the face instead. In which case I would see this?
```
        re = (alpha_current-elem + alpha_neighbor) / 2 * _test[_i][_qp];
```
